packages/vitest/src/node/config.ts

```diff
@@ -92,6 +92,11 @@ export function resolveConfig(
     mode,
   } as ResolvedConfig
 
+  if (resolved.watch) {
+    resolved.coverage.cleanOnRerun = true
+    resolved.coverage.all = false
```

Comment:
Does this override user's configurations, or simply set default values?

I'm not sure if we should override the values, as users are then unable to configure these at all in watch mode.

Reply:
This does overrides the user config, I have modified the changed, after new commits. Thanks for pointing these out.